File: backend/app/models/request.rb
class Request < ApplicationRecord
  STATUSES = %w[
    submitted
    auto_approved
    pending_approval
    approved
    rejected
  ].freeze

  belongs_to :tenant
  belongs_to :request_type
  belongs_to :requester, class_name: "User"

  has_one :approval

  validates :requested_value, numericality: { greater_than: 0 }
  validates :status, inclusion: { in: STATUSES }
end

File: frontend/smart-rule-based-approval-engine/src/models/Request.ts
export interface RequestItem {
    id: string; // The serializer returns id, usually string/number, safe to use string/number
    type: string;
    requested_value: string;
    status: string; // 'submitted' | 'approved' | 'rejected'
    approval: {
        action: string;
        reason: string;
        approver_id: number;
    } | null;
}
